### ISSUE TITLE
Wrong `libsymspg` shared library can be chosen in the new tests (title changed)

### ISSUE THREAD

**Issue by atztogo** [spglib/spglib#182]
I use compilers installed via conda-forge. In this conda env, if I have spglib installed via conda-forge, it seems the new google test takes ~~`$CONDA_PREFIX/include/spglib.h`~~ `$CONDA_PREFIX/lib/libsymspg.*` instead of ~~`src/spglib.h`~~ the shared library locally made, which resulted in the test failures.

**Comment by lan496** [comment]
@atztogo Sorry for my hasty merge. I am trying to reproduce the test failures.

**Comment by lan496** [comment]
@atztogo I cannot still reproduce the test failures...

Adding [`BEFORE` variable](https://cmake.org/cmake/help/latest/command/include_directories.html) may fix the priority,
```cmake
# Line 46
include_directories(BEFORE ${PROJECT_SOURCE_DIR}/../src)
```
but please revert the commit. I will investigate on.

**Comment by atztogo** [comment]
Don't worry. This is a minor issue, but just helpful in case it is easily solved.
 
I got the same issue on m1 mac and x86-64 linux (ubuntu 20.04). The following is my environment (m1 mac):

```bash
conda create -n spglib-test compilers spglib cmake
conda activate spglib-test
ls $CONDA_PREFIX/include/spglib.h
/Users/togo/.miniforge/envs/spglib-test/include/spglib.h
mkdir spglib-test
cd spglib-test
git clone git@github.com:spglib/spglib.git
cd spglib/test/
cmake -S . -B build -DUSE_SANITIZER="Address" -DCMAKE_BUILD_TYPE="Debug"
cmake --build build
cd build
ctest

...


77% tests passed, 6 tests failed out of 26

Total Test time (real) =   1.62 sec

The following tests FAILED:
	 16 - test_dataset_access.test_spg_get_spacegroup_type_from_symmetry (Subprocess aborted)
	 17 - test_magnetic_symmetry.test_spg_get_magnetic_symmetry_from_database (Subprocess aborted)
	 19 - test_magnetic_symmetry.test_spg_get_symmetry_with_site_tensors (Failed)
	 20 - test_magnetic_symmetry.test_spg_get_magnetic_spacegroup_type (Subprocess aborted)
	 21 - test_magnetic_symmetry.test_spg_get_magnetic_dataset (Subprocess aborted)
	 22 - test_magnetic_symmetry.test_spg_get_magnetic_dataset_type4 (Subprocess aborted)
Errors while running CTest
Output from these tests are in: /Users/togo/spglib-test/spglib/test/build/Testing/Temporary/LastTest.log
Use "--rerun-failed --output-on-failure" to re-run the failed cases verbosely.
```

**Comment by atztogo** [comment]
In the case of linux,

```
cmake --build build  

...

[ 83%] Linking CXX executable test_dataset_access
/home/togo/spglib-test/spglib/test/build/test_dataset_access: symbol lookup error: /home/togo/spglib-test/spglib/test/build/test_dataset_access: undefined symbol: spg_get_spacegroup_type_from_symmetry
CMake Error at /home/togo/.miniconda/envs/spglib-test/share/cmake-3.24/Modules/GoogleTestAddTests.cmake:112 (message):
  Error running test executable.

    Path: '/home/togo/spglib-test/spglib/test/build/test_dataset_access'
    Result: 127
    Output:


Call Stack (most recent call first):
  /home/togo/.miniconda/envs/spglib-test/share/cmake-3.24/Modules/GoogleTestAddTests.cmake:225 (gtest_discover_tests_impl)


make[2]: *** [CMakeFiles/test_dataset_access.dir/build.make:117: test_dataset_access] Error 1
make[2]: *** Deleting file 'test_dataset_access'
make[1]: *** [CMakeFiles/Makefile2:350: CMakeFiles/test_dataset_access.dir/all] Error 2
```

But this issue will be resolved when spglib v2.0 is released from conda-forge. So please do not use much time for this issue.

**Comment by atztogo** [comment]
@lan496,  sorry, `spglib.h` was not the reason of the problem, but that is the shared library in `$CONDA_PREFIX/lib` is chosen. With setting `LD_LIBRARY_PATH` could solve the issue. Another solution that may be better is to create static link library instead of the shared library by changing `test/CMakeLists.txt` as follows:
```
#add_library(symspg SHARED ${SOURCES})
add_library(symspg STATIC ${SOURCES})
```
This works for me on m1 mac and x86-64 linux.
